check/src/typecheck.rs

```diff
@@ -146,7 +146,7 @@ impl<'a> KindEnv for Environment<'a> {
             .get(type_name)
             .map(|&(_, ref alias)| {
                 let mut kind = Kind::typ();
-                for arg in alias.args.iter().rev() {
+                for arg in alias.typ.params().iter().rev() {
```

Comment:
Not sure if we should be working off aliases here now. Might we be missing `forall`s?